### PR TITLE
Fixes precision issue when padding nested tensor

### LINKED ISSUE #175866: Padding for Nested Long Tensor loses precision
### 🐛 Describe the bug

The padding for Nested Long Tensor ends to loss of precision.

```python
import torch

sample = nested_tensor([
  tensor([                   1, -7704671799122851728]),
  tensor([-7704671799122851728])
])

padded_tensor = torch.nested.to_padded_tensor(sample, padding=-7704671799122851728)
print(padded_tensor)
# tensor([[                   1, -7704671799122851728],
#         [-7704671799122851728, -7704671799122851840]])

```

As can be seen, the padding value loses precision. Note that rest of the values don't lose precision (e.g. `-7704671799122851728` in the sample don't lose precision but in the padding loses the precision)

### Versions
```
PyTorch version: 2.7.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 26.3 (arm64)
GCC version: Could not collect
Clang version: 21.1.8
CMake version: Could not collect
Libc version: N/A

Python version: 3.11.14 (main, Nov 29 2025, 01:23:00) [Clang 17.0.0 (clang-1700.4.4.1)] (64-bit runtime)
Python platform: macOS-26.3-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Apple M4 Max

Versions of relevant libraries:
[pip3] mypy==1.13.0
[pip3] mypy-boto3-s3==1.39.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] pytest-mypy==0.10.3
[pip3] pytorch-lightning==2.6.1
[pip3] torch==2.7.1
[pip3] torchmetrics==1.8.2
[pip3] tritonclient==2.51.0
[conda] Could not collect
```

cc @cpuhrsch @jbschlosser @bhosmer @drisspg @soulitzer @davidberard98 @YuqingJ

### PR DESCRIPTION
**Pull request by sumitbinnani**
Fixes #175866

The padding for Nested Long Tensor might lead to loss of precision.

```python
import torch

sample = nested_tensor([
  tensor([                   1, -7704671799122851728]),
  tensor([-7704671799122851728])
])

padded_tensor = torch.nested.to_padded_tensor(sample, padding=-7704671799122851728)
print(padded_tensor)
# tensor([[                   1, -7704671799122851728],
#         [-7704671799122851728, -7704671799122851840]])
```
As can be seen, the padding value loses precision. 

_Note that rest of the values don't lose precision (e.g. -7704671799122851728 in the sample don't lose precision but the same value when used for padding loses the precision)_

### CODE REVIEW

**Comment by sumitbinnani** [comment]
https://github.com/pytorch/pytorch/issues/175866#issuecomment-3985385891

@malfet @soulitzer I believe the issue was closed in mistake. The bug is the fact that padding has been defined as float as compared to scalar.

This PR fixes the padding of the nested long tensors.
